bin/warehouse/generate_warehouse.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+"""
+Creates Warehouse for all PD-Types found in archived backups.
+
+Arguments:
+fname - directory of archived backups
+operation - '-d' to compare last 2 backups (default), '-a' to compare all backups.
+"""
+import tarfile
+import sys
+import os
+import subprocess
+import shutil
+from datetime import datetime
+
+fname = sys.argv[1]
+
+if len(sys.argv) == 3:
+    operation = sys.argv[2]
+    if operation != '-a' and operation != '-d':
+        sys.exit("Error: invalid operation value (sys.argv[2]), -d or -a expected")
+else:
+    operation = "-d"
+
+tar_array = sorted(os.listdir(fname))
+if operation == "-d":
+    tar_array = tar_array[-2:]
+print(tar_array)
+prev = ''
+curr = ''
+
+def get_base(tfile):
+    base = os.path.basename(tfile)
+    pd_name = os.path.splitext(os.path.splitext(base)[0])[0]
+    return pd_name
+
+def extract(tfile, dest):
+    fpath = './' + dest
+    tar = tarfile.open(fname + tfile)
+    tar.extractall(path=fpath)
+    tar.close()
+    return fpath
+
+def run_migrations(fpath):
+    if not os.path.exists('temp'):
+        os.mkdir('temp')

```

Comment:
always use the tempfile module to create temporary directories. This avoids security and concurrency issues.